src/webgpu/util/color_space_conversion.ts

```diff
@@ -191,3 +191,68 @@ export function srgbToDisplayP3(pixel: {
 
   return pixel;
 }
+
+type InPlaceColorConversion = (rgba: {
+  R: number;
+  G: number;
+  B: number;
+  readonly A: number; // Alpha never changes during a conversion.
+}) => void;
+
+/**
+ * Returns a function which applies the specified colorspace/premultiplication conversion.
```

Comment:
I'm a bit curious about this. Is this change motivated by better code structure or the performance?

Reply:
Mainly just code structure, since this conversion code had gotten quite large and was in the middle of a large function. And it's a good reference implementation that we probably want to continue using elsewhere.